cmd/podman/registry/config.go

```diff
@@ -103,9 +103,15 @@ func setXdgDirs() error {
 	}
 
 	if _, found := os.LookupEnv("DBUS_SESSION_BUS_ADDRESS"); !found {
-		sessionAddr := filepath.Join(os.Getenv("XDG_RUNTIME_DIR"), "bus")
-		if _, err := os.Stat(sessionAddr); err == nil {
-			os.Setenv("DBUS_SESSION_BUS_ADDRESS", "unix:path="+sessionAddr)
+		runtimeDir := os.Getenv("XDG_RUNTIME_DIR")
+		if runtimeDir != "" {
```

Comment:
This isn't quite the same as mine. Even if you get it to compile, the logic is different: in my approach, I first assemble `XDG/bus`, _then_ check if that exists, _then_ deref links.

`XDG` will always be set. That's some early `config` hackery that @giuseppe added years ago. What does not exist on macs is `XDG + bus`.